plugin/tools.py

```diff
@@ -702,14 +702,61 @@ def get_clang_version_str(clang_binary):
         log.info("Getting version from command: `%s`", check_version_cmd)
         output_text = Tools.run_command(check_version_cmd, shell=True)
 
-        # now we have the output, and can extract version from it
+        if sublime.platform() == "osx":
+            return cls.get_clang_version_str_osx(output_text)
+        elif sublime.platform() == "linux":
+            return cls.get_clang_version_str_linux(output_text)
+        elif sublime.platform() == "windows":
+            return cls.get_clang_version_windows(output_text)
+        else:
+            log.error("Unknown platform: %s" % sublime.platform())
+            return None
+
+    @classmethod
+    def get_clang_version_str_osx(cls, output_text):
+        """Get Clang version string from output of "clang_binary -v".
+
+        This is the OSX-specific variant.
+
+        Args:
+            output_text (str): output of "clang_binary -v"
+
+        Returns:
+            str: clang version number
+
+        Raises: RuntimeError: There is an error while getting version. This is
+            too important to continue. If this fails the plugin will not work
+            at all.
+        """
+        # There can be two flavours on osx: the apple flavour or the trunk/brew
+        # flavour.
+        if "Apple" in output_text:
+            return cls.get_apple_clang_version_str(output_text)
+        else:
+            # Same as on linux
+            return cls.get_clang_version_str_linux(output_text)
+
+    @classmethod
+    def get_apple_clang_version_str(cls, output_text):
+        """Get Clang version string from output of "clang_binary -v".
+
+        This is the AppleClang-specific variant.
+
+        Args:
+            output_text (str): output of "clang_binary -v"
+
+        Returns:
+            str: clang version number
+
+        Raises: RuntimeError: There is an error while getting version. This is
+            too important to continue. If this fails the plugin will not work
+            at all.
+        """
         version_regex = re.compile("\d\.\d\.*\d*")
         match = version_regex.search(output_text)
         if match:
             version_str = match.group()
-            # TODO: we may need to revisit this and detect, say, word apple in
-            # the output of clang --version
-            if version_str >= "4.2" and platform.system() == "Darwin":
+            if version_str >= "4.2":
```

Comment:
We don't need this check now do we? We will only reach this when we know we are on osx and use apple clang, right?